spring-boot-2-demo-app/src/main/java/io/github/mfvanek/spring/boot2/test/service/PublicApiService.java

```diff
@@ -0,0 +1,64 @@
+package io.github.mfvanek.spring.boot2.test.service;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.github.mfvanek.spring.boot2.test.service.dto.CurrentTime;
+import io.github.mfvanek.spring.boot2.test.service.dto.ParsedDateTime;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.TimeZone;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@Getter
+public class PublicApiService {
+
+    @Value("${app.retries}")
+    private String retries;

```

Comment:
Should be `int` and should be used below in `.retryWhen(Retry.fixedDelay(3, Duration.ofSeconds(2))`